api/filters.go

```diff
@@ -257,16 +262,29 @@ func (api *FilterAPI) addFilterJobDimensionOption(w http.ResponseWriter, r *http
 func (api *FilterAPI) updateFilterJob(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 	filterID := vars["filter_job_id"]
+
 	filter, err := models.CreateFilter(r.Body)
 	if err != nil {
 		log.Error(err, log.Data{"filter_job_id": filterID})
 		http.Error(w, badRequest, http.StatusBadRequest)
 		return
 	}
 
+	authenticationHeader := r.Header.Get(internalToken)
+
+	var isAuthenticated bool
+	if authenticationHeader != "" {
+		isAuthenticated, err = api.checkAuthentication(authenticationHeader)
+		if err != nil {
+			log.Error(err, log.Data{"filter_job_id": filterID})
+			setErrorCode(w, err)
+			return
+		}
+	}
+
 	filter.FilterID = filterID
 
-	err = api.dataStore.UpdateFilter(filterID, filter)
+	err = api.dataStore.UpdateFilter(isAuthenticated, filterID, filter)
```

Comment:
Why pass the ID through separately to the filter, when the ID is contained in the filter anyway?